signal/playerctl/playerctl_lib.lua

```diff
@@ -98,15 +98,21 @@ local function metadata_cb(player, metadata)
         if player ~= last_player or title ~= last_title or
            artist ~= last_artist or artUrl ~= last_artUrl
         then
-            awful.spawn.with_line_callback(get_album_art(artUrl), {
-                stdout = function(line)
-                    awesome.emit_signal("bling::playerctl::title_artist_album",
-                                        title,
-                                        artist,
-                                        line,
-                                        player.player_name)
-                end
-            })
+            if artUrl ~= "" then
+                awful.spawn.with_line_callback(get_album_art(artUrl), {
+                    stdout = function(line)
+                        awesome.emit_signal("bling::playerctl::title_artist_album",
+                                            title,
+                                            artist,
+                                            line,
+                                            player.player_name)
+                    end
+                })
+            else
+                awesome.emit_signal(
+                        "bling::playerctl::title_artist_album", title,
+                        artist, player.player_name)
```

Comment:
> This will break client side code because the signal is defined to pass the `artUrl` as the fourth argument to any function connected to it. Receiving the player name in that position instead will most likely cause issues.

The reason I did it was that  if some widget relies on it, and the new media has no artUrl then that widget will keep on showing that old artUrl. I wasn't aware it would break things, but if so there are other ways to fix this issue.

Reply:
Due to the possibility of the artUrl field being empty client side code shouldn't rely on it always being filled in. That's probably something that should be added to the docs. Sorry if I sounded a bit harsh I was just pointing out that even if the artUrl is empty we still need to emit the signal with the artUrl in the correct position and can let the client side code deal with receiving an empty string.

Reply:
The issue is I'm not sure why when the first signal is emitted, a new notification is spawned with the last media artUrl showing. I understand why you would want to send an empty string instead of nothing, but like I said there's some issue with it currently.

Here's my code for the notification, I'm really not doing much:
```
awesome.connect_signal("bling::playerctl::title_artist_album",
                       function(title, artist, art_path)
    naughty.notification({title = title, text = artist, image = art_path})
end)
```